Avatar-Badge

ark2002 opened the pull request:
- added avatar and badges components
- added main.css
- linked the avatar and badge pages from component page
- added a media query for mobile view in component page

ark2002 committed:
feat:  added avatar & badges components

-added avatar and badges components
-added main.css
- linked the avatar and badge pages from component page
-added a media query for mobile view in component page

Mohasin-Haque commented on components/avatar/avatar.css:
```diff
@@ -0,0 +1,63 @@
+  position: relative;
+}
+.avatar__img {
+  width: 57px;
+  height: 57px;
```
Please use relative units as much as you can.

Mohasin-Haque commented on components/avatar/avatar.html:
```diff
@@ -0,0 +1,117 @@
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
```
You can add more meta descriptions to personalise it more. Like
<meta name="author" content="Your name" />
    <meta name="description" content="About your ui" />

ark2002 replied:
hey i have made the suggested changes you can look it up

Mohasin-Haque commented on components/badge/badge.html:
```diff
@@ -0,0 +1,60 @@
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
```
same can be done here also

Mohasin-Haque commented on components/badge/badge.html:
```diff
@@ -0,0 +1,60 @@
+      <nav class="header_nav">
+        <ul>
+          <li><a href="#">Documentation</a></li>
+          <li><a href="/components_page.html"> Components</a></li>
```
Remove the whitespace in component

Mohasin-Haque commented on main.css:
```diff
@@ -0,0 +1,97 @@
+  width: 40px;
+}
+.header__nav {
+  display: flex;
```
You can also make one class for flex as you are using it again and again .. so it will reduce your work also.

Mohasin-Haque reviewed (changes requested):
Your code looks good to me but you need to work on the comments. 
In the preview I saw that you haven't wrote documentation for any of the component.. Please do it. 
Thank you

ark2002 committed:
fix: added meta description

- added meta descriptions to the files
- added flex-row and flex-column classes
-better formatting

yogesh- commented on components/badge/badge.css:
```diff
@@ -0,0 +1,40 @@
+  font-size: 2rem;
+  background-color: var(--primary-color);
+  border-color: transparent;
+  padding: 5px 10px;
+  border-radius: 5px;
+  color: var(--white-color);
+  cursor: pointer;
+}
+
+.badge__demo-txt {
+  margin-right: 0.5rem;
+}
```
Don't use rem and px together. 
Stick to rem and define the root font size to 62.5% which makes 1rem = 10px.

ark2002 replied:
hey i have made the suggested changes can you look it up

xii-vi commented on components/avatar/avatar.css:
```diff
@@ -0,0 +1,62 @@
+  border-radius: 50%;
+}
+.material-icons.avatar__icon {
+  font-size: 40px;
```
Use relative unit.

ark2002 replied:
hey ! its an icon used from the material icons stylesheet so thought using a absolute unit for it was the better way to work with

ark2002 replied:
made the changes

yogesh- commented on components_page.html:
```diff
@@ -1,97 +1,97 @@
+            <span>Alerts</span>
+          </div>
+          <div class="component__img-container">
+            <img src="/assests/alerts.svg" class="component__img" />
```
standard practice, always put alt_text in the image tag

arjunbharti commented on components/avatar/avatar.css:
```diff
@@ -0,0 +1,62 @@
+.busy {
+  background-color: var(--busy-color);
+}
+.notif {
+  background-color: var(--notif-color);
+  text-align: center;
+  font-family: var(--secondary-font-family);
+  font-weight: 600;
+  color: var(--white-color);
+}
```
It would be better if you write proper name just like you did above.

xii-vi reviewed (commented):
Use relative unit if it is greater than 10px.

yogesh- reviewed (changes requested):
Minor changes suggested, otherwise looks good to me.

arjunbharti commented on components/badge/badge.css:
```diff
@@ -0,0 +1,40 @@
+  position: absolute;
+}
+
+.badge--large {
+  font-size: 2.2rem;
+  padding: 2.5px 10px;
+}
+
+.badge--position {
+  margin: -10px;
+}
```
Try to stick to one naming convention. Here you are using '--' and in the above classes you have used '__'
It's bit confusing

ark2002 replied:
i have used '__' for sub-elements of the block while I have used '--' for the specific attributes like sizes , positions etc.

ark2002 committed:
fix: converted all units to relative units

- converted all absolute units to relative units
- changed notif class to notification
- added alt for images

arjunbharti reviewed (approved):
Looks fine now!

xii-vi commented:
It's good now, you can merge
